editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationController.java

```diff
@@ -0,0 +1,310 @@
+          parcelLinesFeatureLayer = new FeatureLayer(geodatabase.getGeodatabaseFeatureTable("ParcelLines_1"));
+
+          // create annotation layers from tables in the geodatabase
+          addressPointsAnnotationLayer = new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("Loudoun_Address_PointsAnno_1"));
+          parcelLinesAnnotationLayer = new AnnotationLayer(geodatabase.getGeodatabaseAnnotationTable("ParcelLinesAnno_1"));
+
+          // add the feature layers to the map
+          map.getOperationalLayers().add(addressPointFeatureLayer);
+          map.getOperationalLayers().add(parcelLinesFeatureLayer);
+
+          // add the annotation layers to the map
+          map.getOperationalLayers().add(addressPointsAnnotationLayer);
+          map.getOperationalLayers().add(parcelLinesAnnotationLayer);
+        } else {
+          // show alert if geodatabase fails to load
+          new Alert(Alert.AlertType.ERROR, "Error loading Geodatabase.").show();
+        }
+      });
+      geodatabase.loadAsync();
+
+      // set on click behaviour
+      mapView.setOnMouseClicked(event -> {
+        // check that the primary mouse button was clicked
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // create a point where the user clicked
+          Point2D screenPoint = new Point2D(event.getX(), event.getY());
+          // call select or move method to move to the point
+          selectOrMove(screenPoint);
+        }
+      });
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Select the nearest feature, or move the point or polyline vertex to the given screen point.
+   */
+  private void selectOrMove(Point2D screenPoint) {
+
+    // if a feature hasn't been selected, select the feature
+    if (selectedFeature == null) {
+      identifyFeature(screenPoint);
+    } else {
+      // convert the screen point to a map point
+      Point mapPoint = mapView.screenToLocation(screenPoint);
+      // if the feature is a polyline, move the polyline
+      if (selectedFeatureIsPolyline) {
+        movePolylineVertex(mapPoint);
+      } else {
+        // if the feature is a point, move the point
+        movePoint(mapPoint);
+      }
+    }
+  }
+
+  /**
+   * Identify a feature near the given screen point.
+   */
+  private void identifyFeature(Point2D screenPoint) {
+
+    // clear any previously selected features
+    clearSelection();
+    selectedFeature = null;
+
+    // identify across all layers
+    ListenableFuture<List<IdentifyLayerResult>> identifyLayerResultsFuture = mapView.identifyLayersAsync(screenPoint, 1, false);
+    identifyLayerResultsFuture.addDoneListener(() -> {
+      try {
+        // get the list of results from the future
+        List<IdentifyLayerResult> identifyLayerResults = identifyLayerResultsFuture.get();
+        // if one or more results have been identified
+        if (identifyLayerResults.size() > 0) {
```

Comment:
It's generally considered better to write 
```suggestion
        if (!identifyLayerResults.size().isEmpty()) {
```